onboard/catkin_ws/src/offboard_comms/Arduino Sketchbooks/ThrusterServoArduino/MultiplexedBasicESC.h

```diff
@@ -1,15 +1,14 @@
-#ifndef BasicESC_h
-#define BasicESC_h
+#ifndef MULTIPLEXEDBASICESC_H
+#define MULTIPLEXEDBASICESC_H
 
 #include <Arduino.h>
-#include "MultiplexedObject.h"
 #include "Adafruit_PWMServoDriver.h"
+#include "MultiplexedServo.h"
 
-class MultiplexedBasicESC:public MultiplexedObject{
+class MultiplexedBasicESC: public MultiplexedServo{
 public:
-    MultiplexedBasicESC(Adafruit_PWMServoDriver *, int);
-    void initialise();
-    void run(int);
+  void attach(uint8_t);
+  void write(int8_t);
```

Comment:
This declaration here actually hides the `write` function in MultiplexedServo, instead of overriding it. It doesn't technically matter here since we don't use the two interchangeably, however, it's not considered good practice.

So for instance, a `MultiplexedBasicESC*` would call the `MultiplexedBasicESC` version of the function, but if you cast it to a `MultiplexedServo*`, it would call the `MultiplexedServo` version, this is typically not how you want the inheritance hierarchy to work.